File: 1861. rotating-the-box.py
class Solution:
    def rotateTheBox(self, box: List[List[str]]) -> List[List[str]]:
        r, c=len(box), len(box[0])
        rotate=[['.']*r for _ in range(c)]
        for i, row in enumerate(box):
            bottom=c-1
            for j in range(c-1, -1, -1):
                if row[j]=='#':
                    rotate[bottom][r-1-i]='#'
                    bottom-=1
                elif row[j]=='*':
                    rotate[j][r-1-i]='*'
                    bottom=j-1
        return rotate


# Test cases
if __name__ == "__main__":
    from typing import List
    
    solution = Solution()
    
    # Example 1
    boxGrid1 = [["#",".","#"]]
    result1 = solution.rotateTheBox(boxGrid1)
    print("Example 1: Input: boxGrid = [[\"#\",\".\",\"#\"]]")
    print(f"Output: {result1}")
    print(f"Expected: [[\".\"], [\"#\"], [\"#\"]]\n")
    
    # Example 2
    boxGrid2 = [["#",".","*","."],["#","#","*","."]]
    result2 = solution.rotateTheBox(boxGrid2)
    print("Example 2: Input: boxGrid = [[\"#\",\".\",\"*\",\".\"],[\"#\",\"#\",\"*\",\".\"]]")
    print(f"Output: {result2}")
    print(f"Expected: [[\"#\",\".\"], [\"#\",\"#\"], [\"*\",\"*\"], [\".\",\".\"]]\n")
    
    # Example 3
    boxGrid3 = [["#","#","*",".","*","."],["#","#","#","*",".","."],["#","#","#",".","#","."]]
    result3 = solution.rotateTheBox(boxGrid3)
    print("Example 3: Input: boxGrid = [[\"#\",\"#\",\"*\",\".\",\"*\",\".\"],[\"#\",\"#\",\"#\",\"*\",\".\",\".\"],[\"#\",\"#\",\"#\",\".\",\"#\",\".\"]]")
    print(f"Output: {result3}")
    print(f"Expected: [[\".\",\"#\",\"#\"], [\".\",\"#\",\"#\"], [\"*\",\"*\",\"*\"], [\"#\",\"*\",\".\"], [\"#\",\".\",\"*\"], [\"#\",\".\",\"\"]]")

File: 1861. rotating-the-box.java
class Solution {
    public char[][] rotateTheBox(char[][] boxGrid) {
        int  n = boxGrid.length;
        int m = boxGrid[0].length;

        for(int i=0; i<n; i++){
            int empty = m - 1;
            for(int j=m-1; j>=0; j--){
                if(boxGrid[i][j] =='*'){
                    empty = j - 1;
                }else if(boxGrid[i][j] == '#'){
                    boxGrid[i][j] = '.';
                    boxGrid[i][empty] = '#';
                    empty--;
                }
            }
        }

        char[][] result = new char[m][n];
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                result[j][n-i-1] = boxGrid[i][j];
            }
        }

        return result;
    }
    
    // Helper function to print 2D char array
    private static void print2DArray(char[][] arr) {
        System.out.print("[");
        for (int i = 0; i < arr.length; i++) {
            System.out.print("[");
            for (int j = 0; j < arr[i].length; j++) {
                System.out.print("\"");
                System.out.print(arr[i][j]);
                System.out.print("\"");
                if (j < arr[i].length - 1) System.out.print(",");
            }
            System.out.print("]");
            if (i < arr.length - 1) System.out.print(",");
        }
        System.out.println("]");
    }
    
    // Test cases
    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Example 1
        char[][] boxGrid1 = {{"#", ".", "#".toCharArray()[0]}};
        boxGrid1[0] = new char[]{'#', '.', '#'};
        char[][] result1 = solution.rotateTheBox(boxGrid1);
        System.out.println("Example 1: Input: boxGrid = [[\"#\",\".\",\"#\"]]\n");
        System.out.print("Output: ");
        print2DArray(result1);
        System.out.println("Expected: [[\".\"], [\"#\"], [\"#\"]]\n");
        
        // Example 2
        char[][] boxGrid2 = {{'#', '.', '*', '.'}, {'#', '#', '*', '.'}};
        char[][] result2 = solution.rotateTheBox(boxGrid2);
        System.out.println("Example 2: Input: boxGrid = [[\"#\",\".\",\"*\",\".\"],[\"#\",\"#\",\"*\",\".\"]]\n");
        System.out.print("Output: ");
        print2DArray(result2);
        System.out.println("Expected: [[\"#\",\".\"], [\"#\",\"#\"], [\"*\",\"*\"], [\".\",\".\"]]\n");
        
        // Example 3
        char[][] boxGrid3 = {{'#', '#', '*', '.', '*', '.'}, {'#', '#', '#', '*', '.', '.'}, {'#', '#', '#', '.', '#', '.'}};
        char[][] result3 = solution.rotateTheBox(boxGrid3);
        System.out.println("Example 3: Input: boxGrid = [[\"#\",\"#\",\"*\",\".\",\"*\",\".\"],[\"#\",\"#\",\"#\",\"*\",\".\",\".\"],[\"#\",\"#\",\"#\",\".\",\"#\",\".\"]]\n");
        System.out.print("Output: ");
        print2DArray(result3);
        System.out.println("Expected: [[\".\",\"#\",\"#\"], [\".\",\"#\",\"#\"], [\"*\",\"*\",\"*\"], [\"#\",\"*\",\".\"], [\"#\",\".\",\"*\"], [\"#\",\".\",\".\"]]\n");
    }
}